apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2556,8 +2558,9 @@ check_tx_and_verify_signatures(SignedTx, Updates, Mod, Data, Pubkeys, ErrTypeMsg
             %% TODO: conduct more relevant checks
             verify_signatures_onchain_check(Pubkeys, SignedTx);
         {aesc_client_reconnect_tx = Mod, Tx} ->
-            try {Mod:channel_pubkey(Tx), Mod:role(Tx), Mod:origin(Tx)} of
-                {ChannelPubkey, MyRole, MyPubkey} ->
+            try {Mod:channel_pubkey(Tx), Mod:role(Tx), Mod:origin(Tx), Mod:round(Tx)} of
+                {ChannelPubkey, MyRole, MyPubkey, Round}
+                  when Round > Data#data.client_reconnect_nonce ->
```

Comment:
Is the uncaught `try_clause` intended?
```
If the evaluation of Exprs succeeds without an exception, the patterns Pattern are sequentially matched against the result in the same way as for a case expression, except that if the matching fails, a try_clause run-time error occurs.
```
```
1> try 2 of Round when Round > 5 -> ok catch error:_ -> {error, invalid} end.
** exception error: no try clause matching 2
```

Reply:
I see that the only point where this function is called with reconnect is:
```
    try check_client_reconnect_tx(Tx, D) of
    ...
    catch
        error:E ->
            lager:debug("CAUGHT ~p / ~p", [E, erlang:get_stacktrace()]),
            keep_state(D, [{reply, From, E}])
    end
```
so seems ok even if counterintuitive - as same function returns either error tuple of exception.

Reply:
It happens to catch class error hence works...
```
1> try begin try 2 of Round when Round > 5 -> ok catch error:_ -> {error, invalid} end end of _ -> ok catch error:_ -> caught end.
caught
```

Reply:
Addressed in new commit